src/api/entities/Portfolio/index.ts

```diff
@@ -212,6 +216,91 @@ export abstract class Portfolio extends Entity<UniqueIdentifiers, HumanReadable>
     return values(assetBalances);
   }
 
+  /**
+   * Retrieve the NFTs held in this portfolio
+   *
+   *  @param args.assets - array of NftCollection (or tickers) for which to fetch holdings (optional, all holdings are retrieved if not passed)
+   */
+  public async getNftsHeld(args?: {
+    assets: (string | NftCollection)[];
+  }): Promise<PortfolioNftHolding[]> {
+    const {
+      owner: { did },
+      _id: portfolioId,
+      context: {
+        polymeshApi: {
+          query: { portfolio },
+        },
+      },
+      context,
+    } = this;
+
+    const rawPortfolioId = portfolioIdToMeshPortfolioId({ did, number: portfolioId }, context);
+    const [exists, freeBalanceEntries, lockedBalanceEntries] = await Promise.all([
+      this.exists(),
+      portfolio.portfolioNFT.entries(rawPortfolioId),
+      portfolio.portfolioLockedNFT.entries(rawPortfolioId),
+    ]);
+
+    if (!exists) {
+      throw new PolymeshError({
+        code: ErrorCode.DataUnavailable,
+        message: notExistsMessage,
+      });
+    }
+
+    const queriedCollections = args?.assets.map(asset => asTicker(asset));
+    const seenTickers = new Set<string>();
+
+    const processBalance = (
+      entry: (typeof freeBalanceEntries)[0],
+      balanceRecord: Record<string, Nft[]>
+    ): void => {
+      const [key] = entry;
+
+      const ticker = tickerToString(key.args[1][0]);
+      const heldId = u64ToBigNumber(key.args[1][1]);
```

Comment:
```suggestion
      const [
        {
          args: [, [rawTicker, rawNftId]],
        },
      ] = entry;

      const ticker = tickerToString(rawTicker);
      const heldId = u64ToBigNumber(rawNftId);
```
It feels cleaner this way instead of referencing via indices